### DISCUSSION TITLE
Accessing Podman Desktop from WSL

### DISCUSSION THREAD

**Discussion by YulawOne** [nix-community/NixOS-WSL#463] [Q&A]
Hello everyone! 

I'm quite new to NixOs, and I cannot figure out how to set up Podman connection to [Podman Desktop](https://podman-desktop.io/docs/podman/accessing-podman-from-another-wsl-instance)

I found a configuration for docker-desktop, however, there is no similar setup for Podman and additionally, I don't see a way how to install the podman-remote package on Nixos.

If anyone came across the issue please help 🙏

**Comment by DwcksDinh** [comment]
I have same issue with you

**Comment by ian-h-chamberlain** [comment]
I was able to get this working by installing Podman desktop on Windows, then following the instructions you linked.

> I don't see a way how to install the podman-remote package on Nixos.

As it turns out, you don't need to! From that page:
> The Podman client is available with a full podman installation or with the podman-remote version 4.x and higher. On Ubuntu it is generally easier to install podman-remote.

So basically, I think you just need something like this for your config, and `nixos-rebuild switch` (you might also need to restart WSL for the group to take effect):
```nix
    pkgs = with pkgs; [
      podman
    ];
    users.groups.podman = {
      gid = 10;
      members = [ yourUserName ];
    };
```

Then I did
```
podman system connection add --default podman-machine-default-root unix:///mnt/wsl/podman-sockets/podman-machine-default/podman-root.sock
```
and everything started working.

The only caveat I've found so far is you need to use `podman --remote` or `podman -r` for every command, which is slightly annoying but can be worked around with a shell alias easily enough, I guess...
